server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -1094,38 +1094,33 @@ public static Settings resolveSettings(ComposableIndexTemplate template, Map<Str
     }
 
     /**
-     * Resolve the given v2 template into an ordered list of aliases
-     *
-     * @param failIfTemplateHasDataStream Whether to skip validating if a template has a data stream definition and an alias definition.
-     *                                    This validation is needed so that no template gets created that creates data stream and also
-     *                                    an alias pointing to the backing indices of a data stream. Unfortunately this validation
-     *                                    was missing in versions prior to 7.11, which mean that there are cluster states out there
-     *                                    that have this malformed templates. This method is used when rolling over a data stream
-     *                                    or creating new data streams. In order for these clusters to avoid failing these operations
-     *                                    immediately after an upgrade the failure should be optional. So that there is time to change
-     *                                    these templates. The logic that adds/updates index and component templates shouldn't skip this
-     *                                    validation.
+     * Resolve the given v2 template name into an ordered list of aliases
```

Comment:
@martijnvg, could you confirm that the `failIfTemplateHasDataStream` parameter is no longer necessary here? I believe it was used only to verify that data stream templates do not create aliases which is no longer true.